### PR TITLE
1222 my communities

### LINKED ISSUE #1222: Overview of communities you maintain
As a user, you should have access to an overview page where you can see which communities you maintain. This should be called "My communities", just like "My software" etc.

### PR DESCRIPTION
**Pull request by dmijatovic**
# Add my communities to user section

Closes #1222

Changes proposed in this pull request:
* My communities added to user menu option if communities module enabled in settings.json
* Communities page added to users section. It uses community list layout
* My software page updated to use software list items
* My projects page updated to use project list items
* My organisations page uses organisation list (new component created)
* Project metadata page moved lover in the user menu (on the right side of page)
   
How to test:
* enable communities module in frontend/public/data/settings.json
```json
{
  "host":{
     ...
     "modules":["software","projects","organisations","communities"]
   }
   ...
}
```
* `make start` to build app and generate test data 
* login as rsd admin (first user is by default admin in dev)
* navigate to software overview, select software, edit software. Navigate to maintainers section, create link and copy/pate link in the address-bar to become maintainer of this software. Add yourself as maintainer to few other software entries
* navigate to projects overview, select project, edit project. Add yourself as maintainer using invite link (as you did with software)
* navigate to organisations overview, select organisation. Navigate to settings and invite yourself to be maintainer. Do this with few organisations. 
* navigate to communities overview, select community. Navigate to settings and invite yourself as maintainer. Do this with multiple communities. 

## Example my software
image

## Example my projects
image

## Example my organisations
image

## Example my communities
image

PR Checklist:

*   [ ] Increase version numbers in `docker-compose.yml`
*   [x] Link to a GitHub issue
*   [ ] Update documentation
*   [x] Tests

### CODE REVIEW

File: frontend/components/user/communities/useUserCommunities.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+export async function getCommunitiesForMaintainer({searchFor,page,rows,account,token}:GetCommunitiesProps){
+  try{
+    // all top level communities of maintainer
+    const query=`maintainer_id=${account}&order=software_cnt.desc.nullslast,name`
+    // baseUrl
+    let url =`${getBaseUrl()}/rpc/communities_by_maintainer?${query}`
+    // search
+    if (searchFor) {
+      url += `&name=ilike.*${searchFor}*`
```
Comment (ewan-escience) [inline frontend/components/user/communities/useUserCommunities.tsx:35]: Please URL encode `searchFor`, also in other places where applicable.

**Review by ewan-escience — CHANGES_REQUESTED** [review]
Nice work! I have one comment in the code and the following suggestion:

The number of rows shown are not saved, maybe you can reuse the cookie value of `rsd_page_rows`?

**Comment by jmaassen** [comment]
For projects, organizations and communities, the list view also shows the image, but for software it does not?

**Comment by dmijatovic** [comment]
@jmaassen Well spotted. It is fixed now. Can you approve. please?

**Review by ewan-escience — APPROVED** [review]
On "My communities", the counts sometimes show as `0–0 of 0`, but this will be fixed soon.

**Pull request merged**